### ISSUE TITLE
Helm upgrade fails

### ISSUE THREAD

**Issue by iJebus** [TIBCOSoftware/js-docker#71]
Running `helm upgrade`, including with unchanged install values, throws the error seen below.

```sh
helm upgrade -i jrs jrs/helm --namespace jrs --timeout 17m0s --debug
history.go:56: [debug] getting history for release jrs
upgrade.go:139: [debug] preparing upgrade for jrs
upgrade.go:147: [debug] performing update for jrs
upgrade.go:319: [debug] creating upgraded release for jrs
client.go:218: [debug] checking 8 resources for changes
client.go:501: [debug] Looks like there are no changes for ServiceAccount "jasperserver-pro"
client.go:501: [debug] Looks like there are no changes for Secret "jasperserver-license"
client.go:501: [debug] Looks like there are no changes for Role "jasperserver-role"
client.go:501: [debug] Looks like there are no changes for RoleBinding "jasperserver-role"
client.go:501: [debug] Looks like there are no changes for Service "jasperserver-cache-service"
client.go:501: [debug] Looks like there are no changes for Service "jrs-jasperserver-pro"
client.go:250: [debug] error updating the resource "jrs-jasperserver-pro":
         failed to create patch: map: map[] does not contain declared merge key: name
upgrade.go:420: [debug] warning: Upgrade "jrs" failed: failed to create patch: map: map[] does not contain declared merge key: name
Error: UPGRADE FAILED: failed to create patch: map: map[] does not contain declared merge key: name
helm.go:88: [debug] failed to create patch: map: map[] does not contain declared merge key: name
helm.sh/helm/v3/pkg/kube.(*Client).Update
        helm.sh/helm/v3/pkg/kube/client.go:263
helm.sh/helm/v3/pkg/action.(*Upgrade).releasingUpgrade
        helm.sh/helm/v3/pkg/action/upgrade.go:365
runtime.goexit
        runtime/asm_amd64.s:1581
UPGRADE FAILED
main.newUpgradeCmd.func2
        helm.sh/helm/v3/cmd/helm/upgrade.go:200
github.com/spf13/cobra.(*Command).execute
        github.com/spf13/cobra@v1.2.1/command.go:856
github.com/spf13/cobra.(*Command).ExecuteC
        github.com/spf13/cobra@v1.2.1/command.go:974
github.com/spf13/cobra.(*Command).Execute
        github.com/spf13/cobra@v1.2.1/command.go:902
main.main
        helm.sh/helm/v3/cmd/helm/helm.go:87
runtime.main
        runtime/proc.go:255
runtime.goexit
        runtime/asm_amd64.s:1581
```

**Comment by iJebus** [comment]
Updating `deployment.yaml` as below resolves the issue. I'm not sure if there's a better way of doing it though.

```yaml
    spec:
      {{- if .Values.image.PullSecrets }}
      imagePullSecrets:
        - name: {{ .Values.image.PullSecrets }}
      {{- end }}
```

**Comment by rreddyv-tibco** [comment]
@iJebus  Yes pull secret causing the issue as it is empty by default.  It will be fixed in subsequent releases with better approach

**Comment by rreddyv-tibco** [comment]
It has been fixed in 8.0.2 release and hence closing the issue
